task-maker-async/src/client.rs

```diff
@@ -430,6 +433,233 @@ fn prepare_execution_group(
     Some(ret)
 }
 
+// TODO(veluca): consider using the tokio versions of file io.
+async fn read_file_to_memory(
+    store: &StoreClient,
+    file: &FileHandle,
+    size_limit: Option<usize>,
+) -> Result<Vec<u8>, Error> {
+    let mut result = vec![];
+    loop {
+        if let Some(limit) = size_limit {
+            if result.len() >= limit {
+                break;
+            }
+        }
+        let chunk = store
+            .read_chunk(context::current(), *file, result.len())
+            .await??;
+        match chunk {
+            FileReadingOutcome::Dropped => {
+                result.clear();
+            }
+            FileReadingOutcome::EndOfFile => {
+                break;
+            }
+            FileReadingOutcome::Data(chunk) => {
+                result.extend(chunk);
+            }
+        };
+    }
+    Ok(result)
+}
+
+async fn read_file_to_disk(
+    store: &StoreClient,
+    file: &FileHandle,
+    destination: &Path,
+    make_executable: bool,
+) -> Result<(), Error> {
+    std::fs::create_dir_all(destination.parent().unwrap())?;
+    let mut destination = std::fs::File::create(destination)?;
+    if make_executable {
+        use std::os::unix::fs::PermissionsExt;
+        destination.set_permissions(Permissions::from_mode(0o755))?;
+    }
+    loop {
+        let chunk = store
+            .read_chunk(
+                context::current(),
+                *file,
+                destination.stream_position()? as usize,
+            )
+            .await??;
+        match chunk {
+            FileReadingOutcome::Dropped => {
+                destination.set_len(0)?;
+                destination.seek(io::SeekFrom::Start(0))?;
```

Comment:
Same as before, if the file is dropped should we try again from the start?

Reply:
That's what this code does ;)

Reply:
Yeah, but it was basically the same question as above. And I assume it has the same answer as well.